.github/workflows/gem-publish.yml

```diff
@@ -0,0 +1,46 @@
+name: Publish Ruby Gem
+
+on:
+  push:
+    branches: [ master ]
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    name: Build and Publish
```

Comment:
should we check semaphore before publish?

Reply:
So there are two options here:
1. enforce branches are upto date so the branch is the same as master, so tests have already run on semaphore
2. we add a check to the pipeline

What would you prefer @pawelma @AlanGabbianelli ?

Reply:
@addersuk I think adding a check to the pipeline is more explicit.

Reply:
So having looked at this we don't have a secret available in GitHub Actions for the Semaphore API and the risk is that API would have access to all our repos, not just this one.

We can use the GitHub status api, but that is not aware of branches 🤦 , so it can cause flaky problems that a branch runs on the same sha and is cancelled for some reason, this would then cause deploys to fail.

Would it be okay to enforce up to date with master before merging to keep these issue simple for now.  If that causes issues we can review at a later date.

Reply:
Sounds good to me @addersuk. Let's keep it simple. 😉